src/signals/incident/components/form/MapSelectors/Asset/Selector/DetailPanel/DetailPanel.tsx

```diff
@@ -1,312 +1,183 @@
 // SPDX-License-Identifier: MPL-2.0
-// Copyright (C) 2021 - 2022 Gemeente Amsterdam, Vereniging van Nederlandse Gemeenten
-import { useCallback, useState, useContext, useRef, useEffect } from 'react'
-import type { KeyboardEvent, ChangeEvent, FC } from 'react'
+import type { FC } from 'react'
+// Copyright (C) 2021 - 2023 Gemeente Amsterdam, Vereniging van Nederlandse Gemeenten
+import { useCallback, useContext, useMemo, useState } from 'react'
 
 import { ChevronLeft } from '@amsterdam/asc-assets'
-import {
-  Label,
-  Input,
-  Checkbox,
-  Button,
-  breakpoint,
-  ascDefaultTheme,
-} from '@amsterdam/asc-ui'
+import { ascDefaultTheme, breakpoint } from '@amsterdam/asc-ui'
 import { useDispatch } from 'react-redux'
 import { useMediaQuery } from 'react-responsive'
 
 import { formatAddress } from 'shared/services/format-address'
 import type { PdokResponse } from 'shared/services/map-location'
-import {
-  selectionIsObject,
-  selectionIsUndetermined,
-  UNKNOWN_TYPE,
-  NEARBY_TYPE,
-  UNREGISTERED_TYPE,
-} from 'signals/incident/components/form/MapSelectors/constants'
+import { selectionIsObject } from 'signals/incident/components/form/MapSelectors/constants'
 import { closeMap } from 'signals/incident/containers/IncidentContainer/actions'
 
 import {
-  AddressPanel,
   Description,
-  LegendToggleButton,
-  OptionsList,
   PanelContent,
   StyledAssetList,
+  StyledBackButton,
   StyledButton,
+  StyledButtonWrapper,
   StyledLabelPDOkAutoSuggest,
-  StyledLegendPanel,
-  StyledPDOKAutoSuggest,
+  StyledParagraphPDOkAutoSuggest,
 } from './styled'
+import { useResetDrawerState } from './useResetDrawerState'
+import {
+  DrawerOverlay,
+  DrawerState,
+} from '../../../../../../../../components/DrawerOverlay'
 import AssetSelectContext from '../../context'
-import { ScrollWrapper, Title } from '../styled'
+import Legend from '../Legend'
+import { ScrollWrapper, StyledPDOKAutoSuggest } from '../styled'
 
 export interface DetailPanelProps {
   language?: Record<string, string>
+  zoomLevel?: number
 }
 
-const nearbyLegendItem = {
-  label: 'Bestaande melding',
-  icon: {
-    iconUrl: '/assets/images/area-map/icon-pin.svg',
-    iconSize: [40, 40],
-  },
-  typeValue: NEARBY_TYPE,
-}
-
-/* istanbul ignore next */
-const DetailPanel: FC<DetailPanelProps> = ({ language = {} }) => {
-  const shouldRenderAddressPanel = useMediaQuery({
-    // Set breakpoint to mobile instead of tablet since desktop will get the mobile version when zoom on 200% which is not accesible with keyboard.
-    query: breakpoint('max-width', 'mobileL')({ theme: ascDefaultTheme }),
+const DetailPanel: FC<DetailPanelProps> = ({ language, zoomLevel }) => {
+  const [drawerState, setDrawerState] = useState<DrawerState>(DrawerState.Open)
+  const [legendOpen, setLegendOpen] = useState(false)
+  const shouldRenderMobileVersion = useMediaQuery({
```

Comment:
Any reason not to use ASC's [useMatchMedia](https://github.com/Amsterdam/amsterdam-styled-components/blob/main/packages/asc-ui/src/utils/hooks/useMatchMedia.ts) for this?

Reply:
Not sure no